modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -0,0 +1,260 @@
+        else:
+            logger.warning(
+                "Number of parallel prefetch requests not explicitly given in training config - defaulting to 1."
+            )
+            parallel_prefetch_requests = 1
+
+        if "tokenizer" in data_config:
+            tokenizer = data_config["tokenizer"]
+        else:
+            tokenizer = None
+
+        if "transformations" in data_config:
+            transform_list = data_config["transformations"]
+        else:
+            transform_list = []
+
+        self.dataloader_info = DataLoaderInfo(
+            self.pipeline_id,
+            dataset_id=data_config["dataset_id"],
+            num_dataloaders=training_config["dataloader_workers"],
+            batch_size=training_config["batch_size"],
+            bytes_parser=data_config["bytes_parser_function"],
+            transform_list=transform_list,
+            storage_address=f"{self.modyn_config['storage']['hostname']}:{self.modyn_config['storage']['port']}",
+            selector_address=f"{self.modyn_config['selector']['hostname']}:{self.modyn_config['selector']['port']}",
+            num_prefetched_partitions=num_prefetched_partitions,
+            parallel_prefetch_requests=parallel_prefetch_requests,
+            tokenizer=tokenizer,
+        )
+
+    def _init_model_downloader(self) -> None:
+        assert self.pipeline_config is not None
+        assert self.modyn_config is not None
+
+        self.model_downloader = ModelDownloader(
+            self.modyn_config,
+            self.pipeline_id,
+            self.pipeline_config["training"]["device"],
+            self.base_dir,
+            f"{self.modyn_config['model_storage']['hostname']}:{self.modyn_config['model_storage']['port']}",
+        )
+
+    def _create_dirs(self) -> None:
+        assert self.pipeline_id is not None
+        assert self.base_dir is not None
+
+        self.exp_output_dir = self.base_dir / str(self.pipeline_id) / "datadrift"
+        self.drift_dir = self.exp_output_dir / "drift_results"
+        os.makedirs(self.drift_dir, exist_ok=True)
+
+    def init_trigger(self, pipeline_id: int, pipeline_config: dict, modyn_config: dict, base_dir: pathlib.Path) -> None:
+        self.pipeline_id = pipeline_id
+        self.pipeline_config = pipeline_config
+        self.modyn_config = modyn_config
+        self.base_dir = base_dir
+
+        if len(self.trigger_config) > 0:
+            self._parse_trigger_config()
+
+        self._init_dataloader_info()
+        self._init_model_downloader()
+        self._create_dirs()
+
+    def run_detection(self, reference_embeddings_df: pd.DataFrame, current_embeddings_df: pd.DataFrame) -> bool:
+        assert self.dataloader_info is not None
+        assert self.drift_dir is not None
+
+        # Run Evidently detection
+        column_mapping = ColumnMapping(embeddings={self.evidently_column_mapping_name: reference_embeddings_df.columns})
```

Comment:
Can you add a comment on what the column mapping name is?

Reply:
done